packages/solana/src/providers/WalletConnectProvider.ts

```diff
@@ -0,0 +1,254 @@
+import UniversalProvider from '@walletconnect/universal-provider'
+import { SolConstantsUtil, type AnyTransaction, type Chain, type Provider } from '../utils/scaffold'
+import { ProviderEventEmitter } from './shared/ProviderEventEmitter'
+import type { SessionTypes } from '@walletconnect/types'
+import base58 from 'bs58'
+import {
+  Connection,
+  PublicKey,
+  Transaction,
+  VersionedTransaction,
+  type SendOptions
+} from '@solana/web3.js'
+import { isVersionedTransaction } from '@solana/wallet-adapter-base'
+
+export type WalletConnectProviderConfig = {
+  provider: UniversalProvider
+  chains: Chain[]
+}
+
+export class WalletConnectProvider extends ProviderEventEmitter implements Provider {
+  public readonly name = 'WalletConnect'
+  public readonly type = 'WALLET_CONNECT'
+  public readonly icon = 'https://avatars.githubusercontent.com/u/37784886'
```

Comment:
Do we fetch this from Cloud? @enesozturk

Reply:
This was already in use before https://github.com/WalletConnect/web3modal/blob/main/packages/solana/src/client.ts#L421, but we may change it if is better.

Reply:
@zoruka Can we use this one: https://imagedelivery.net/_aTEfDRm7z3tKgu9JhfeKA/05338e12-4f75-4982-4e8a-83c67b826b00/md (from our CDN service)